WordPress/gutenberg
# Framework: Support registering and invoking actions in the data module

The same way we support registering and calling selectors, this PR adds support for registering and calling actions to the data module.

 - We now have a `wp.data.dispatch` function taking a key and an action (similar to select)
 - We have a `wp.data.registerActions` to register the exposed actions.
 - We also have support for `mapDispatchToProps` in the `query` HoC.

I implemented inserting a block as an exposed action in this PR.

As a follow-up, we could provide a `registerState` helper function to register everything at once: `reducer, actions and selectors`.

**Testing instructions**

 - Open the console of your browser
 - Type `wp.data.dispatch( 'core/editor' ).insertBlocks( wp.blocks.createBlock( 'core/image' ) );` in your browser.
 - A new image block should be created.
